videoInputSrcAndDemos/libs/videoInput/videoInput.cpp

```diff
@@ -854,6 +881,25 @@ int videoInput::listDevices(bool silent){
 			        continue;  // Skip this one, maybe the next one will work.
 			    }
 
+				// Find unique name
```

Comment:
Thanks for this! 
Feels like the below section of code you added should only be called if the FriendlyName query is successful. 
That way the uniqueName is added to the deviceUniqueNames vector when the deviceCount is increased on successful friendly name query.   

Otherwise feels like there could be a danger of the device count and the deviceUniqueNames size not matching. 
Feel free to - update the PR and I'll merge this in. 

Thanks!

Reply:
@ofTheo now I think we should return to this 8c70e9badca882454cc3bbec1a8592a0129c5a10 initial version of PR.. because in 928bfb8ae95d3133d81b52583166dc1b25799732 this method wll not work properly:

https://github.com/ofTheo/videoInput/blob/870dbabda79470736e5d5294038ae5676ebb238a/videoInputSrcAndDemos/libs/videoInput/videoInput.cpp#L812

The main goal is even if FrendlyName query is failed, we should add string to deviceUniqueNames query to match deviceNames indicies.

deviceNames is vector of FriendlyNames and is initialized with 255 empty strings:

https://github.com/ofTheo/videoInput/blob/870dbabda79470736e5d5294038ae5676ebb238a/videoInputSrcAndDemos/libs/videoInput/videoInput.cpp#L770

So we have initially 255 devices whose FriendlyNames are updated after listDevices() is called. So we just skip device in deviceNames if FriendlyName query is failed leaving it empty, but uniqueDeviceNames not increased for this device, and we have index mismatch.